GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs

```diff
@@ -125,52 +118,52 @@ private bool LaunchInsideSpinner(Func<bool> method, string message)
                 null);
         }
 
-        private bool TryLoadUpgradeRing(out ProductUpgrader.RingType ring, out string consoleError)
+        private bool TryInitialize(out string errorMessage)
         {
-            bool loaded = false;
-            if (!this.upgrader.TryLoadRingConfig(out consoleError))
-            {
-                EventMetadata metadata = new EventMetadata();
-                metadata.Add("Upgrade Step", nameof(this.TryLoadUpgradeRing));
-                metadata.Add("Load Error", consoleError);
-                this.tracer.RelatedError(metadata, $"{nameof(this.TryLoadUpgradeRing)} failed.");
-                this.ExitCode = ReturnCode.GenericError;
-            }
-            else
-            {
-                consoleError = null;
-                loaded = true;
-            }
-            
-            ring = this.upgrader.Ring;
-            return loaded;
+            return this.upgrader.Initialize(out errorMessage);
         }
 
         private bool TryRunUpgrade(out Version newVersion, out string consoleError)
         {
             newVersion = null;
 
             Version newGVFSVersion = null;
-            GitVersion newGitVersion = null;
-            string errorMessage = null;
+            string error = null;
+            string consoleMessage;
+            bool isError;
+
+            if (!this.upgrader.CanRunUsingCurrentConfig(out isError, out consoleMessage, out error))

```

Comment:
Why do we need to have both an out string and an out bool here? Can we not always show `error` if it's not null or empty?

Reply:
right, we don't need both. Will remove it.